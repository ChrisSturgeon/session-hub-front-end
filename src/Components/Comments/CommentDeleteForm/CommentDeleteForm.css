.comment-delete.activate-btn,
.comment-delete.confirm-cancel .cancel-btn {
  color: grey;
  font-size: 0.7rem;
  border: 1px solid var(--border-grey);
  border-radius: 5px;
  cursor: pointer;
}

.comment-delete.activate-btn:hover,
.comment-delete.confirm-cancel .cancel-btn:hover {
  background-color: rgb(222, 222, 222);
  /* color: white; */
  border: 1px solid rgb(222, 222, 222);
}

.comment-delete.activate-btn span,
.comment-delete.confirm-cancel span {
  display: flex;
  align-items: center;
  gap: 0.25em;
}

.comment-delete.confirm-cancel {
  font-size: 0.7rem;
  display: flex;
}

.comment-delete.confirm-cancel .confirm-btn {
  color: white;
  background-color: var(--dark-blue);
  border: 1px solid var(--dark-blue);
  border-radius: 5px;
  cursor: pointer;
}
