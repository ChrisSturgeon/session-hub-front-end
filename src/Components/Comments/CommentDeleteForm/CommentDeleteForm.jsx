import './CommentDeleteForm.css';
import { useState } from 'react';
import { APIURL } from '../../../api';

export default function CommentDeleteForm({ commentID, setCommentIsVisible }) {
  const [formVisible, setFormVisible] = useState(false);

  async function handleSubmit(event) {
    event.preventDefault();
    const url = `${APIURL}/comments/${commentID}`;

    try {
      const response = await fetch(url, {
        method: 'DELETE',
        mode: 'cors',
        headers: {
          'Content-Type': 'application/json',
          Authorization: `bearer ${window.localStorage.getItem('JWT')}`,
        },
      });

      if (response.status === 200) {
        setCommentIsVisible(false);
      } else {
        console.log('An error has occured');
      }
    } catch (err) {
      console.log(err);
    }
  }

  if (!formVisible) {
    return (
      <button
        onClick={() => setFormVisible(true)}
        className="comment-delete activate-btn"
      >
        <span>
          <ion-icon name="trash-outline"></ion-icon>
          <div>Remove</div>
        </span>
      </button>
    );
  }

  return (
    <div className="comment-delete confirm-cancel">
      <form onSubmit={(event) => handleSubmit(event)}>
        <button className="confirm-btn">Confirm</button>
      </form>
      <button onClick={() => setFormVisible(false)} className="cancel-btn">
        Cancel
      </button>
    </div>
  );
}
