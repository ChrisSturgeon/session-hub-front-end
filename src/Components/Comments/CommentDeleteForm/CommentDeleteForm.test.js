import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import CommentDeleteForm from './CommentDeleteForm';

describe('Comment Delete Form', () => {
  render(<CommentDeleteForm />);
  it('Activation button renders', () => {
    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  it('Displays form on activation button click', () => {
    render(<CommentDeleteForm />);
    const activationButton = screen.getByRole('button');
    userEvent.click(activationButton);

    expect(screen.getByRole('form')).toBeInTheDocument();
  });

  it('Cancel button hides form', () => {
    render(<CommentDeleteForm />);
    const activationButton = screen.getByRole('button');
    userEvent.click(activationButton);
    const cancelButton = screen.getByRole('button', { name: /Cancel/i });
    userEvent.click(cancelButton);

    expect(screen.queryByRole('form')).not.toBeInTheDocument();
  });
});
