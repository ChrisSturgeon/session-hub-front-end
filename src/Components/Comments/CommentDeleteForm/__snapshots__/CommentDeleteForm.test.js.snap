// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Delete Form Matches snapshot 1`] = `
<div>
  <button
    class="comment-delete activate-btn"
  >
    <span>
      <ion-icon
        name="trash-outline"
      />
      <div>
        Remove
      </div>
    </span>
  </button>
</div>
`;
